File: 169. Majority Element/Solution.java
class Solution {
    public int majorityElement(int[] nums) {
        int majority = nums[0];
        int count = 0;

        for(int i=0;i<nums.length;i++){
            if(count == 0){
                count = 1;
                majority = nums[i];
            }else if(majority != nums[i])
                count--;
            else count++;
        }

        return majority;
    }
}

File: 169. Majority Element/Solution.cpp
#include <vector>

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int majority = nums[0], count = 0;

        for(int i : nums)
        {
            if(i == majority) count++;
            else count--;

            if(count == 0)
            {
                count =  1;
                majority = i;
            }
        }
        return majority;
    }
};